indico/modules/events/papers/operations.py

```diff
@@ -81,3 +86,24 @@ def close_cfp(event):
     event.cfp.close()
     logger.info("Call for papers for %r closed by %r", event, session.user)
     event.log(EventLogRealm.management, EventLogKind.negative, 'Papers', 'Call for papers closed', session.user)
+
+
+@no_autoflush
+def judge_paper(contribution, contrib_data, judgment, judge, send_notifications=False):
+
+    if judgment == PaperAction.accept:
+        contribution.paper_last_revision.state = PaperRevisionState.accepted
+    elif judgment == PaperAction.reject:
+        contribution.paper_last_revision.state = PaperRevisionState.rejected
+    elif judgment == PaperAction.to_be_corrected:
+        contribution.paper_last_revision.state = PaperRevisionState.to_be_corrected
+    if contrib_data['judgment_comment']:
```

Comment:
you can always set the comment. that way judging it again (even though this shouldn't happen) would properly overwrite an existing comment